pkg/controllers/user/multiclusterapp/controller.go

```diff
@@ -0,0 +1,267 @@
+package multiclusterapp
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	corev1 "github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	pv3 "github.com/rancher/types/apis/project.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/cache"
+)
+
+const (
+	globalScopeAnswersKey     = "global"
+	creatorIDAnn              = "field.cattle.io/creatorId"
+	multiClusterAppIDSelector = "mClusterApp"
+	projectIDFieldLabel       = "field.cattle.io/projectId"
+	multiClusterAppPrefix     = "mcapp-"
+)
+
+type MCAppController struct {
+	AppClient             pv3.AppInterface
```

Comment:
the convention is to not use *client in names. Should be `apps` instead. Listers are fine.